xls/passes/select_simplification_pass.cc

```diff
@@ -964,6 +966,35 @@ absl::StatusOr<bool> SimplifyNode(Node* node, const QueryEngine& query_engine,
     }
   }
 
+  // Specialize the arms of the select based on the selector value which chooses
+  // the arm.
+  if (node->Is<Select>()) {
+    Select* sel = node->As<Select>();
+    int64 selr_width = sel->selector()->BitCountOrDie();
+    bool changed = false;
+    for (int64 selr_value = 0; selr_value < sel->cases().size(); ++selr_value) {
+      Node* arm = sel->cases()[selr_value];
+      // Skip any cases where the select arm is used other than just in the
+      // select statement
+      if (arm->users().size() != 1) {
+        continue;
+      }
+      for (Node* selr_use : sel->selector()->users()) {
+        if (!post_dominator_analysis.NodeIsPostDominatedBy(selr_use, arm)) {
+          continue;
+        }
+        auto literal = std::make_unique<Literal>(
```

Comment:
Use Function::MakeNode which automatically adds the node to the function and verifies it:

XLS_ASSIGN_OR_RETURN(Literal* literal, node->function()->MakeNode<Literal>(...));

Reply:
Sounds good, wasn't sure if it was OK to add a Literal if it never gets referenced within the Function